scene/main/canvas_item.cpp

```diff
@@ -120,7 +96,7 @@ void CanvasItem::set_visible(bool p_visible) {
 		return;
 	}
 
-	_propagate_visibility_changed(p_visible);
+	_propagate_visibility_changed(p_visible, !p_visible);
```

Comment:
Seems unnecessary to pass both `p_visible` and `!p_visible`, you can compute the latter in the method directly.

Reply:
Not really. The second argument can be `true` only in this single case, every other `_propagate_visibility_changed()` call should pass false. I had to make this change to have `hidden` signal emitted in correct order.